tsdb/store.go

```diff
@@ -957,6 +946,20 @@ func IsRetryable(err error) bool {
 	return true
 }
 
+// DecodeStorePath extracts the database and retention policy names
+// from a given shard or WAL path.
+func DecodeStorePath(shardOrWALPath string) (database, retentionPolicy string) {
```

Comment:
I assumed `filepath` was better suited to the job since it handles any unusual cases involving a doubled-up separator, e.g. `/some/path//db/rp/...`. Maybe it's not possible to hit that situation in our code paths, but I chose to err on the side of being defensive.